app/routes.py

```diff
@@ -1 +1,167 @@
+    db.session.commit()
+
+    return {
+        "details": f'Task {task.id} "{task.title}" successfully deleted'
+    }, 200
+
+@tasks_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def mark_incomplete(task_id):
+    task = validate_model(Task, task_id)
+
+    task.completed_at = None
+
+    db.session.commit()
+
+    return {"task": task.to_json()}, 200
+
+@tasks_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def mark_complete(task_id):
+    task = validate_model(Task, task_id)
+
+    task.completed_at = datetime.utcnow()
+    db.session.commit()
+    
+    send_slack_message(Task, task)
+
+    return {"task": task.to_json()}, 200
+
+
+@goals_bp.route("", methods=["POST"])
+def create_goal():    
+    new_goal = Goal.from_json(request.get_json())
+
+    db.session.add(new_goal)
+    db.session.commit()
+
+    return {"goal": new_goal.to_json()}, 201
+
+@goals_bp.route("", methods=["GET"])
+def get_all_goals():
+    goals = filter_and_sort(Goal)
+        
+    goals_response = [goal.to_json() for goal in goals]
+    
+    return jsonify(goals_response), 200
+
+@goals_bp.route("/<goal_id>", methods=["GET"])
+def get_goal_by_id(goal_id):
+    goal = validate_model(Goal, goal_id)
+    
+    return {"goal": goal.to_json()}, 200
+
+@goals_bp.route("/<goal_id>", methods=["PUT"])
+def update_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+
+    request_body = request.get_json()
+    
+    goal.title = request_body["title"],
+    
+    db.session.commit()
+
+    return {"goal": goal.to_json()}, 200
+
+@goals_bp.route("/<goal_id>", methods=["DELETE"])
+def delete_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+
+    db.session.delete(goal)
+    db.session.commit()
+    
+    return jsonify({
+        "details": f"Goal {goal.id} \"{goal.title}\" successfully deleted"
+        }), 200
+
+@goals_bp.route("/<goal_id>/tasks", methods=["POST"])
+def sending_tasks_to_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+
+    request_body = request.get_json()
+
+    for task_id in request_body["task_ids"]:
+        task = validate_model(Task, task_id)
+        task.goal_id = goal.id
+
+    db.session.commit()
+
+    return {
+        "id": goal.id,
+        "task_ids": request_body["task_ids"]
+        }, 200
+
+@goals_bp.route("/<goal_id>/tasks", methods=["GET"])
+def get_tasks_by_goal_id(goal_id):
+    goal = validate_model(Goal, goal_id)
+
+    tasks_response = [task.to_json() for task in goal.tasks]
+
+    return jsonify({
+        "id": goal.id,
+        "title": goal.title,
+        "tasks": tasks_response
+    }), 200
+
+@goals_bp.route("/<goal_id>/mark_complete", methods=["PATCH"])
+def mark_goal_complete(goal_id):
+    goal = validate_model(Goal, goal_id)
+
+    for task in goal.tasks:
+        if not task.completed_at:
+            task.completed_at = datetime.utcnow()
+    
+    db.session.commit()
+    
+    send_slack_message(Goal, goal)
+    return {"goal": goal.to_json()}, 200
```

Comment:
Well done on this project, I didn't have much to comment on and that is a good thing! Keep up the good work! Really looking forward to what you create in the frontend! Please feel free to reach out if you have any questions about the feedback that I left! ✨💫🤭 You killed it, Syd! Really!